target/linux/ath79/dts/qca9556_mikrotik_routerboard-wap-g-5hact2hnd.dts

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "mikrotik,routerboard-wap-g-5hact2hnd", "qca,qca9556";
+	model = "MikroTik RouterBOARD wAP G-5HacT2HnD";
+
+	aliases {
+		serial0 = &uart;
+		mdio-gpio1 = &mdio2;
+		label-mac-device = &eth1;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	mdio2: mdio {
+		compatible = "virtual,mdio-gpio";
+
+		gpios = <&gpio 12 GPIO_ACTIVE_HIGH>, /* MDC */
+			<&gpio 11 GPIO_ACTIVE_HIGH>; /* MDIO */
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		phy1: ethernet-phy@1 {
+			reg = <0>;
+			device_type = "ethernet-phy";
+
+			phy-mode = "sgmii";
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <17>;
+		phy-mode = "rgmii-id";
+	};
+
+	switch0@1f {
+		compatible = "qca,ar8327";
+		reg = <0x1f>;
+		qca,ar8327-initvals = <
+			0x04 0x87600000 /* PORT0 PAD MODE CTRL */
+			0x0c 0x00080080 /* PORT6 PAD MODE CTRL */
+			0x10 0x81000080 /* POWER_ON_STRIP */
+
+			0x7c 0x0000007e /* PORT0_STATUS */
+			0x94 0x0000007e /* PORT6 STATUS */
+			>;
+	};
+};
+
+&mdio1 {
+	status = "okay";
+
+	compatible = "virtual,mdio-gpio";
+
+	phy2: ethernet-phy@2 {
+		reg = <2>;
+		phy-mode = "sgmii";
+	};
+};
+
+&eth0 {
+	status = "disabled";
```

Comment:
Would `compatible = "syscon", "simple-mfd";` be better here?

Reply:
If you use `compatible = "syscon", "simple-mfd";`, you can drop the status.

Reply:
I checked the documentation at mfd.txt and I don't really understand what it means. :(

Anyway, I added the line and tried; everything seems to be working fine. But, if eth0 is disabled, does it make a difference to add it? Just wondering.

Reply:
I've also never understood that properly, but in older support patches I was asked to do this for similar cases, e.g.
https://github.com/openwrt/openwrt/pull/1843#issuecomment-476980265
If you want a proper explanation, maybe ask 981213 ...

Reply:
> If you use `compatible = "syscon", "simple-mfd";`, you can drop the status.
> I've also never understood that properly, but in older support patches I was asked to do this for similar cases, e.g.
#1843 (comment)

OK :)